Client/Frontend/Browser/BrowserLongPressGestureRecognizer.swift

```diff
@@ -0,0 +1,135 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+import Foundation
+import WebKit
+import UIKit
+
+enum ElementType {
+    case ImageSrc(NSURL)
+    case hrefLink(NSURL)
```

Comment:
I don't think we're anticipating any additional elements types, so rather having an array of enums, I think it would be cleaner to use a dictionary of type <code>[LongPressElementType: NSURL]</code> instead. Please remove the NSURL part of these cases and rename them to <code>Image</code> and <code>Link</code> instead.

So, in summary:
```
enum LongPressElementType {
    case Image
    case Link
}
````